severity.go

```diff
@@ -0,0 +1,51 @@
+package log
+
+import (
+	"fmt"
+	"strings"
+	"sync/atomic"
+)
+
+type Severity int32
+
+const (
+	SeverityInfo Severity = iota
+	SeverityWarning
+	SeverityError
+)
+
+var severityName = map[Severity]string{
+	SeverityInfo:    "INFO",
+	SeverityWarning: "WARN",
+	SeverityError:   "ERROR",
+}
```

Comment:
This wasn't part of your change, but while you're here I would make this a slice instead of a map, for much faster lookups:

```go
var severityName = []string{
```

You'll need to change the String() function too:

```go
func (s Severity) String() string {
    return severityName[s]
}
```

And you'll need to cast the result in `severityFromString`:

```go
return Severity(k), nil
```